materialize/resources/resource_connection.go

```diff
@@ -0,0 +1,838 @@
+package resources
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+)
+
+func Connection() *schema.Resource {
+	return &schema.Resource{
+		Description: "The connection resource allows you to manage connections in Materialize.",
+
+		CreateContext: resourceConnectionCreate,
+		ReadContext:   resourceConnectionRead,
+		UpdateContext: resourceConnectionUpdate,
+		DeleteContext: resourceConnectionDelete,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "The name of the connection.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"schema_name": {
+				Description: "The identifier for the connection schema.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Default:     "public",
+			},
```

Comment:
Just merged in but I would add a `database_name` value to fully qualify the resource.